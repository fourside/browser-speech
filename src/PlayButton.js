import React from 'react';

export default class PlayButton extends React.Component {
  render() {
    return (
      <div className="form-group">
        <button className="btn btn-lg btn-primary" type="button" onClick={(e) => this.props.onPlayClick(e)}>Play</button>
      </div>
    );
  }
}

