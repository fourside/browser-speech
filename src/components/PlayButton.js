// @flow
import React from 'react';

type Props = {
  label: string,
  disabled: boolean,
  onPlayClick: (e: Event) => void
};

export default class PlayButton extends React.Component<Props> {
  render() {
    return (
      <button className="btn btn-lg btn-primary btn-play" type="button"
        disabled={this.props.disabled}
        onClick={(e) => this.props.onPlayClick(e)}>
        {this.props.label}
      </button>
    );
  }
}

