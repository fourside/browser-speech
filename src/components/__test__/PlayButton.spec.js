import React from 'react';
import PlayButton from '../PlayButton';
import renderer from 'react-test-renderer';

describe('PlayButton', () => {
  it('should be rendered.', () => {
    const tree = renderer
      .create(<PlayButton label="test" disabled={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

