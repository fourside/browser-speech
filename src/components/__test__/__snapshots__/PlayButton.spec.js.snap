// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton should be rendered. 1`] = `
<button
  className="btn btn-lg btn-primary btn-play"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  test
</button>
`;
